text/2978-stack_based_vec.md

```diff
@@ -176,49 +181,13 @@ impl<T, const N: usize> ArrayVec<T, N> {
 
     pub fn split_off(&mut self, at: usize) -> Self;
 
-    pub fn swap_remove(&mut self, idx: usize) -> T;  // Panics at run-time
+    pub fn swap_remove(&mut self, idx: usize) -> Option<T>;
```

Comment:
`swap_remove` uses a specific index, in this case I feel like panicking is actually better

Reply:
for removal, I think just copying `Vec` 1 by 1 is the correct choice

Reply:
A "safe" `swap_remove` version is similar to `get` in the sense that both return an optional value when out-of-bounds. On the opposite direction, `pub fn swap_remove(&mut self, idx: usize) { ... }` is similar to indexing, e.g., `let value = slice[10];`.

In certain cases, users that don't want to panic at run-time will have to manually check if the index is within bounds before issuing a removing command. If a "panicking" `swap_remove` is indeed desirable, please also consider the inclusion of `try_swap_remove`.

Reply:
I wonder if we could check this at compile time to guarantee `idx < N`.